nni/algorithms/compression/pytorch/quantization/quantizers.py

```diff
@@ -338,6 +430,50 @@ def quantize(self, input_ri, q_bits):
         output = torch.round(input_ri * scale) / scale
         return output
 
+    def export_model(self, model_path, calibration_path=None, onnx_path=None, input_shape=None, device=None):
+        """
+        Export quantized model weights and calibration parameters(optional)
+
+        Parameters
+        ----------
+        model_path : str
+            path to save quantized model weight
+        calibration_path : str
+            (optional) path to save quantize parameters after calibration
+        onnx_path : str
+            (optional) path to save onnx model
+        input_shape : list or tuple
+            input shape to onnx model
+        device : torch.device
+            device of the model, used to place the dummy input tensor for exporting onnx file.
+            the tensor is placed on cpu if ```device``` is None
+        """
+        assert model_path is not None, 'model_path must be specified'
+        self._unwrap_model()
+        calibration_config = {}
+
+        for name, module in self.bound_model.named_modules():
+            if hasattr(module, 'weight_bit'):
+                calibration_config[name] = {}
+                calibration_config[name]['weight_bit'] = int(module.weight_bit)
```

Comment:
so this quantizer does not calibrate activation?

Reply:
In our current implementation of **Dorefa**, it does not quantize activation, so we don't need to calibrate activation.

Reply:
could you double check the paper, does the paper mention how to calibrate activation?

Reply:
After discussion, we reach an agreement that the refactor of `Dorefa` should starts after survey and it will be done in another PR. What' s more, ut related to `export_model()` has been added into code.